src/controllers/util.rs

```diff
@@ -28,6 +28,30 @@ impl AuthenticatedUser {
 impl<'a> UserAuthenticationExt for dyn RequestExt + 'a {
     /// Obtain `AuthenticatedUser` for the request or return an `Unauthorized` error
     fn authenticate(&self, conn: &PgConnection) -> AppResult<AuthenticatedUser> {
+        let forwarded_host = self.headers().get("x-forwarded-host");
+        let forwarded_proto = self.headers().get("x-forwarded-proto");
+        let expected_origin = match (forwarded_host, forwarded_proto) {
```

Comment:
I propose having a comma separated list of origins in an environment variable like `WEB_ALLOWED_ORIGINS=http://localhost:8888,http://localhost:4200`. Any `Origin` headers in the request would need to be on this list. The example above could be added to `.env.sample` so that either testing approach works.

Alternatively, you could fall back to `self.host()` if the other headers aren't set. Note that for this approach you don't want to use `self.scheme()` because that is for the local socket. You'd probably want something like `if self.app().config.env == crate::Env::Production { "https" } else { "http" }`.